pool/connection_pool.go

```diff
@@ -1120,9 +1143,26 @@ func (p *ConnectionPool) fillPools() bool {
 
 		connOpts := p.connOpts
 		connOpts.Notify = end.notify
-		conn, err := tarantool.Connect(addr, connOpts)
+		conn, err := tarantool.Connect(ctx, addr, connOpts)
 		if err != nil {
 			log.Printf("tarantool: connect to %s failed: %s\n", addr, err.Error())
+			select {
+			case <-ctx.Done():
+				ctxCanceled = true
+
+				p.addrs[addr] = nil
+				log.Printf("tarantool: operation was canceled")
+
+				for address, endpoint := range p.addrs {
+					if endpoint == nil {
+						continue
+					}
+					p.deactivateConnection(address, endpoint.conn, endpoint.role)
+				}
+
+				return false, ctxCanceled
+			default:
+			}
 		} else if conn != nil {
```

Comment:
It there still possible now? conn == nil && err == nil.

Reply:
Yes, you are right, it is not possible. Updated.